src/inmanta/module.py

```diff
@@ -753,6 +783,98 @@ def parse(cls, source: Union[str, TextIO]) -> Mapping[str, object]:
                 raise InvalidMetadata(msg="Metadata file doesn't have a metadata section.") from e
 
 
+class PreservativeYamlParser:
+    """
+    A Yaml parser that tries to preserve comments and the order of elements.
+    This parser doesn't preserve indentation.
+    """
+
+    @classmethod
+    def _get_parser(cls) -> YAML:
+        parser = YAML()
+        # Make sure the indentation settings are used consistently
+        parser.indent(mapping=2, sequence=4, offset=2)
+        return parser
+
+    @classmethod
+    def parse(cls, filename: str) -> CommentedMap:
+        parser = cls._get_parser()
+        with open(filename, "r", encoding="utf-8") as fd:
+            return parser.load(fd)
+
+    @classmethod
+    def dump(cls, filename: str, content: CommentedMap) -> None:
+        parser = cls._get_parser()
+        with open(filename, "w", encoding="utf-8") as fd:
+            parser.dump(content, stream=fd)
+
+
+class RequirementsTxtParser:
+    """
+    Parser for a requirements.txt file
+    """
+
+    @classmethod
+    def parse(cls, filename: str) -> List[Requirement]:
+        """
+        Parse a requirements.txt file into a list of Requirements.
+        """
+        with open(filename, "r", encoding="utf-8") as fd:
+            lines = fd.readlines()
+            result = [Requirement.parse(r) for r in lines if r.strip(string.whitespace)]
+            return result
+
+    @classmethod
+    def parse_raw(cls, filename: str, remove_dep_on_pkg: str) -> str:
+        """
+        Returns the content of the requirements.txt file with the dependency on `remove_dep_on_pkg` removed.
+        """
+        result = ""
+        with open(filename, "r", encoding="utf-8") as fd:
+            for line in fd.readlines():
+                if line.strip(string.whitespace) and Requirement.parse(line).key != remove_dep_on_pkg:
+                    result += line
+        return result
+
+
+class RequirementsTxtFile:
```

Comment:
It might be interesting to use this for `ModuleLike._get_requirements_txt_as_list()`. Wdyt?